src/inatinqperf/adaptors/faiss_adaptor.py

```diff
@@ -113,33 +106,23 @@ def _build_flat_index(
         metric: Metric,
         dim: int,
         dataset: HuggingFaceDataset,
+        batch_size: int,
     ) -> faiss.Index:
         if metric in (Metric.INNER_PRODUCT, Metric.COSINE):
-            base = faiss.IndexFlatIP(dim)
+            metric_type = faiss.METRIC_INNER_PRODUCT
         else:
-            base = faiss.IndexFlatL2(dim)
+            metric_type = faiss.METRIC_L2
 
-        index = faiss.IndexIDMap2(base)
```

Comment:
I got rid of `IndexIDMap2` since it doesn't make sense for a FLAT index. Please provide a failing unit test as a counter-argument.

Reply:
If the ID mapping is indeed required, please create an issue and assign it to me.